module/userProjects/helper.js

```diff
@@ -2280,6 +2325,359 @@ module.exports = class UserProjectsHelper {
         })
     }
 
+    /**
+     * generate project certificate.
+     * @method
+     * @name generateCertificate 
+     * @param {Object} data - project data for certificate creation data.
+     * @returns {JSON} certificate details.
+    */
+
+     static generateCertificate(data) {
```

Comment:
@VISHNUDAS-tunerlabs - Need much more comments on all 3 functions newly added. Please refer function - _updateUserProfileBasedOnUserRoleInfo  in file - https://github.com/project-sunbird/ml-survey-service/blob/release-5.1.0/module/observations/helper.js